plugins/module/create_kafka_acl_binding.py

```diff
@@ -0,0 +1,253 @@
+    
+    kafka_mgmt_config.access_token = token["access_token"]
+    
+    def get_kafka_mgmt_client():
+        with rhoas_kafka_mgmt_sdk.ApiClient(kafka_mgmt_config) as kafka_mgmt_api_client:
+            # Create an instance of the API class
+            kafka_mgmt_api_instance = default_api.DefaultApi(kafka_mgmt_api_client)
+            return kafka_mgmt_api_instance
+        
+    def retry_get_admin_url(kafka_mgmt_api_instance, retries = 10, backoff_in_seconds = 6):
+        attempt = 0
+        # Check for kafka_admin_url to be used to create topic
+        while (result['kafka_admin_url'] == ""):
+            try:
+                # Enter a context with an instance of the API client
+                    kafka_id = module.params['kafka_id'] # str | The ID of record
+
+                    # example passing only required values which don't have defaults set
+                    try:
+                        kafka_mgmt_api_response = kafka_mgmt_api_instance.get_kafka_by_id(kafka_id)
+                        kafka_mgmt_api_response
+                        result['kafka_admin_url'] = kafka_mgmt_api_response['admin_api_server_url']
+                        result['kafka_admin_resp_obj'] = kafka_mgmt_api_response.to_dict()
+                        
+                    except rhoas_kafka_mgmt_sdk.ApiException as e:
+                        print("Exception when calling DefaultApi->get_kafka_by_id: %s\n" % e) 
+                        result['message'] = e
+                        module.fail_json(msg='Failed to create kafka instance', **result)
+            except:
+                if attempt == retries:
+                    raise
+                sleep = (backoff_in_seconds * attempt)
+                time.sleep(sleep)
+                attempt += 1
+                
+    # Check for kafka_admin_url to be used to create topic
+    if (module.params['kafka_admin_url'] is None) or (module.params['kafka_admin_url'] == ""):
+        print('kafka_admin_url is not provided')
+        retry_get_admin_url(get_kafka_mgmt_client())
+            
+
+    configuration.host = result['kafka_admin_url']
+    configuration.access_token = token["access_token"]
+    with rhoas_kafka_instance_sdk.ApiClient(configuration) as api_client:
+        api_instance = acls_api.AclsApi(api_client)
+        
+        # some light validation as all the params are required to be in all caps 
+        if module.params['resource_type'] is not None:
+            rt = art(module.params['resource_type'].upper())
+        if module.params['pattern_type'] is not None:
+            pt = apt(module.params['pattern_type'].upper())
+        if module.params['operation_type'] is not None:
+            op = aot(module.params['operation_type'].upper())
+        if module.params['permission_type'] is not None:
+            per = apert(module.params['permission_type'].upper())
+        if (module.params['principal'] is not None) and (module.params['principal'].startswith('User:') is False):
```

Comment:
Missing validation - Principal should not be none.

Reply:
validation for `principal` is handled in the `run_module` function, the `module_args` has `principal` as `required = True` this means that if the principal is not passed, the module will fail

Reply:
I meant that this is checking state that should never happen. Dead code

Reply:
yea, you're right 👍